### DISCUSSION TITLE
Is there a straightforward way to change `<title>` of the document dynamically?

### DISCUSSION THREAD

**Discussion by GabenGar** [remix-run/react-router#11749] [Q&A]
I mostly care about these situations:
1. The route changes and therefore gets a series of changes from base title (while it's fetching data) to final title when it's done. I presume It has to be done somewhere in the loader. But then I can't use hooks within loaders.
2. The route doesn't change but I need to change the title to reflect some status change on the page.

[Related upcoming feature](https://react.dev/blog/2024/04/25/react-19#support-for-metadata-tags) of React itself looks interesting, but it's only an eventual future, not now.

The things I looked up either recommend [`react-helment`](https://github.com/nfl/react-helmet) or wiring up your own hook. The former wasn't updated for 4 years (which is basically a millennium in javascript land) and the latter is going to be ripe with bugs, especially since I'd have to write synchronization logic which should work inside and outside of react rendering tree.

**Comment by tatsmaki** [comment]
https://developer.mozilla.org/en-US/docs/Web/API/Document/title

```
useEffect(() => {
    document.title = "newTitle";
}, [])
```

And you can use `react-helmet` as long as class components are supported in react and w3c doesn't change the `document.title` standard. 
> There is no plans to remove class components.

This shouldn't be a problem in the next 4 years 😄 
 If it works, let it work - react v19 still has class components.
